### PR TITLE
GH-186: Adds tutorial for ccloud produce & consume

### PR DESCRIPTION
**Pull request by rspurgeon**

### CODE REVIEW

File: _data/tutorials.yaml [hunk 1]
```diff
@@ -426,6 +425,15 @@ count-messages:
   introduction: "Messages are stored in Kafka topics until they are removed by Kafka. It can be useful to know how many messages there are currently in a topic. In this example we'll take a topic of pageview data and see how we can count all of the messages in the topic."
   status:
     ksql: enabled
+
+ccloud-produce-consume:
+  title: "How to produce and consume records from Confluent Cloud using ccloud CLI"
+  meta-description: "How to produce and consume records from Confluent Cloud using ccloud CLI"
+  slug: "/ccloud-produce-consume"
+  question: "How do I use the ccloud CLI to produce and consume records from a Confluent Cloud Kafka topic"
```
Comment (ybyzek) [inline _data/tutorials.yaml:433]: `Confluent Cloud Kafka` -- not permitted by the ASF

File: _data/tutorials.yaml [hunk 2]
```diff
@@ -426,6 +425,15 @@ count-messages:
...
+  introduction: "In this tutorial you learn how to use the ccloud CLI to produce and consume records from a Kafka topic in <a href=\"https://confluent.cloud\">Confluent Cloud</a>. This can help you debug your Confluent Cloud based event streaming applications."
```
Comment (ybyzek) [inline _data/tutorials.yaml:434]: This is useful not just for debugging but also getting started and in production

File: _includes/tutorials/ccloud-produce-consume/kafka/markup/dev/setup-ccloud.adoc [hunk 3]
```diff
@@ -0,0 +1,21 @@
+We recommend you run this tutorial in a new Confluent Cloud environment so it doesn't interfere with your other work, and the easiest way to do this is to use the `ccloud-stack` utility.
+This provisions a new Confluent Cloud stack with a new environment, a new service account, a new Kafka cluster and associated credentials, enables Schema Registry and associated credentials, ACLs with wildcard for the service account, and a local configuration file with all above connection information.
+For more information on `ccloud-stack`, read the link:https://github.com/confluentinc/examples/blob/latest/ccloud/ccloud-stack/README.md[documentation].
```
Comment (ybyzek) [inline _includes/tutorials/ccloud-produce-consume/kafka/markup/dev/setup-ccloud.adoc:3]: Please update link to docs page

**Review by ybyzek — COMMENTED** [review]
@rspurgeon could you please include the link to the staging build so PR reviewers can see the rendered tutorial?

**Comment by rspurgeon** [comment]
> @rspurgeon could you please include the link to the staging build so PR reviewers can see the rendered tutorial?

Does this require a specific manual build action to produce?  I looked in the build website semaphore section and it's not obviously available.  Do we do this for each PR?  If so, can we turn on the stage build automatically?

File: _data/tutorials.yaml (hunk 2 above)
Comment (rspurgeon) [reply _data/tutorials.yaml]: I added "build and debug".  I'm not sure about using ccloud produce in production scenarios but i'm open to a suggested revision.

File: _data/tutorials.yaml (hunk 1 above)
Comment (rspurgeon) [reply _data/tutorials.yaml]: Revised to remove Kafka, but not sure if you have another preferred wording

**Comment by ybyzek** [comment]
Note missing `index.html` modifications to include this tutorial (as we discussed in Slack, just didn't want to forget).

**Comment by rspurgeon** [comment]
> Note missing `index.html` modifications to include this tutorial (as we discussed in Slack, just didn't want to forget).

Added

**Review by bbejeck — APPROVED** [review]
Thanks for the tutorial @rspurgeon.  I have some comments for your consideration, but otherwise this LGTM.

File: _includes/tutorials/ccloud-produce-consume/kafka/markup/dev/setup-ccloud.adoc [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+
+Get the open source library link:https://github.com/confluentinc/examples/blob/latest/utils/ccloud_library.sh[ccloud_library.sh] which has functions to interact with Confluent Cloud, including `ccloud-stack`.
+
++++++
+<pre class="snippet"><code class="shell">{% include_raw tutorials/ccloud-produce-consume/kafka/code/tutorial-steps/dev/get-ccloud-library.sh %}</code></pre>
++++++
+
+Create your stack of Confluent Cloud resources by running the following command.
+Set `CLUSTER_CLOUD` and `CLUSTER_REGION` as needed (defaults are shown below).
```
[the remark this answers is not in the document]
Comment (rspurgeon) [reply _includes/tutorials/ccloud-produce-consume/kafka/markup/dev/setup-ccloud.adoc:13]: @bbejeck I did copy this from the other current existing ccloud tutorial (https://kafka-tutorials.confluent.io/kafka-connect-datagen-ccloud/kafka.html).  Do you think it's worth it to break out here, and there as well?

File: _includes/tutorials/ccloud-produce-consume/kafka/markup/dev/setup-ccloud.adoc [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+NOTE: To avoid unexpected charges, carefully evaluate the cost of resources before launching the script and ensure all resources are destroyed after you are done running the tutorial.
+
++++++
+<pre class="snippet"><code class="shell">{% include_raw tutorials/ccloud-produce-consume/kafka/code/tutorial-steps/dev/run-ccloud-stack.sh %}</code></pre>
++++++
+
```
[the remark this answers is not in the document]
Comment (rspurgeon) [reply _includes/tutorials/ccloud-produce-consume/kafka/markup/dev/setup-ccloud.adoc:20]: Thanks @bbejeck , Added this to latest

File: _includes/tutorials/ccloud-produce-consume/kafka/markup/dev/setup-ccloud.adoc (hunk 4 above)
[the remark this answers is not in the document]
Comment (bbejeck) [reply _includes/tutorials/ccloud-produce-consume/kafka/markup/dev/setup-ccloud.adoc:13]: It's up to you.

**Pull request merged**